### PR TITLE
Statistics Endpoints

### PR DESCRIPTION
**Pull request by dborski**
## Description
Adds two student statistics endpoints
-`api/v1/students/:student_id/average_score/`
  -  Returns the average score of all lessons taken by a single student

-`api/v1/lessons/:lesson_id/average_score/`
  - Returns the combined average score of every student's score on an individual lesson

Completes User Story #2 

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which maintains existing functionality)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
## RSpec Results
- 100%

### CODE REVIEW

File: api/models.py [hunk 1]
```diff
@@ -45,6 +46,14 @@ class LessonStudent(models.Model):
...
   updated_at = models.DateTimeField(auto_now=True)
 
+  def student_average_score(student_id):
+    score = LessonStudent.objects.filter(student_id=student_id).aggregate(Avg('score'))
+    return score
+
+  def lesson_average_score(lesson_id):
+    score = LessonStudent.objects.filter(lesson_id=lesson_id).aggregate(Avg('score'))
+    return score
+
```
Comment (Lithnotep) [inline api/models.py:56]: Great job working with aggregates and getting used to this ORM. looks really clean

Comment (melatran) [reply api/models.py:56]: Glad to know models function similar to the models in rails. This looks great

File: api/views.py [hunk 2]
```diff
@@ -86,6 +94,26 @@ def post(self, request, pk):
...
+class LessonAverage(APIView):
+  parser_classes = [JSONParser]
+
+  def get(self, request, pk):
+    average_score = LessonStudent.lesson_average_score(pk)['score__avg']
+    lesson_score = LessonScore(lesson_id=pk, average_score=average_score)
+    
+    return Response(LessonScoreSerializer(lesson_score).data)
+
+
```
Comment (Lithnotep) [inline api/views.py:116]: Great job here, it was straightforward and simple. Tracking back through the serializer and model method helped me put this all together.

**Review by Lithnotep — APPROVED** [review]

File: api/popos.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class StudentScore:
+  def __init__(self, student_id, average_score):
+    self.student_id = student_id
+    self.average_score = average_score
+
+class LessonScore:
+  def __init__(self, lesson_id, average_score):
+    self.lesson_id = lesson_id
+    self.average_score = average_score
```
Comment (melatran) [inline api/popos.py:9]: Woo hoo! A popos file!

File: api/tests/test_models.py [hunk 4]
```diff
@@ -59,4 +59,48 @@ def test_string_representation(self):
...
+  def test_student_average_score(self):
+    score = LessonStudent.student_average_score(self.student2.id)['score__avg']
+
+    self.assertEqual(score, 42.0)
+
+  def test_lesson_average_score(self):
+    score = LessonStudent.lesson_average_score(self.lesson1.id)['score__avg']
+
+    self.assertEqual(score, 76.0)
+
```
Comment (melatran) [inline api/tests/test_models.py:105]: This test looks great

**Review by melatran — APPROVED** [review]

**Pull request merged**